test/fit_tests/common/fit_common.py

```diff
@@ -0,0 +1,528 @@
+        return {'json':result_data.json(), 'text':result_data.text,
+                'status':result_data.status_code,
+                'headers':result_data.headers.get('content-type'),
+                'timeout':False}
+
+
+# Get the list of BMC IP addresses that we can find
+def get_bmc_ips():
+    idlist = [] # list of unique dcmi node IDs
+    # If we have already done this, use that list
+    if len(BMC_LIST) == 0:
+        ipscan = remote_shell('arp')['stdout'].split()
+
+        for ipaddr in ipscan:
+            if ipaddr[0:3] == "172" and remote_shell('ping -c 1 -w 5 ' + ipaddr)['exitcode'] == 0:
+                # iterate through all known IPMI users
+                for item in GLOBAL_CONFIG['credentials']['bmc']:
+                    return_code = remote_shell('ipmitool -I lanplus -H ' + ipaddr + ' -U ' + item['username'] \
+                                               + ' -P ' + item['password'] + ' -R 1 -N 3 dcmi get_mc_id_string')
+                    if return_code['exitcode'] == 0 and return_code['stdout'] not in idlist:
+                        bmc_info = {"ip": ipaddr, "user": item['username'], "pw": item['password']}
+                        idlist.append(return_code['stdout'])
+                        BMC_LIST.append(bmc_info)
+                        break
+        if VERBOSITY >= 6:
+            print "get_bmc_ips: "
+            print "**** BMC IP node count =", len(BMC_LIST), "****"
+
+    return len(BMC_LIST)
+
+# power on/off all compute nodes in the stack via the BMC
+def power_control_all_nodes(state):
+    if state != "on" and state != "off":
+        print "power_control_all_nodes:  invalid state " + state
+        return
+
+    # Get the list of BMCs that we know about
+    node_count = get_bmc_ips()
+
+    # Send power on/off to all of them
+    for bmc in BMC_LIST:
+        return_code = remote_shell('ipmitool -I lanplus -H ' + bmc['ip'] \
+                                   + ' -U ' + bmc['user'] + ' -P ' \
+                                   + bmc['pw'] + ' -R 1 -N 3 chassis power ' + state)
+        if return_code['exitcode'] != 0:
+            print "Error powering " + state + " node: " + bmc['ip']
+
+    return node_count
+
+def mongo_reset():
+    # clears the Mongo database on ORA to default, returns 0 if successful
+    remote_shell('service onrack-conductor stop')
+    remote_shell('/opt/onrack/bin/monorail stop')
+    remote_shell("mongo pxe --eval 'db.dropDatabase\\\(\\\)'")
+    remote_shell('rm -f /var/lib/dhcp/dhcpd.leases')
+    remote_shell('rm -f /var/log/onrack-conductor-event.log')
+    remote_shell('/opt/onrack/bin/monorail start')
+    if remote_shell('service onrack-conductor start')['exitcode'] > 0:
+        return 1
+    return 0
+
+def appliance_reset():
+    return_code = subprocess.call("ipmitool -I lanplus -H " + ARGS_LIST["bmc"] \
+                                  + " -U root -P 1234567 chassis power reset", shell=True)
```

Comment:
Why hard code the user/password here? The RackHD appliance may setup in different hardwares.